gufe/tests/test_custom_json.py

```diff
@@ -115,9 +116,28 @@ def test_round_trip(self):
             json_str = json.dumps(obj, cls=encoder)
             reconstructed = json.loads(json_str, cls=decoder)
             npt.assert_array_equal(reconstructed, obj)
+            assert reconstructed.dtype == obj.dtype
             json_str_2 = json.dumps(obj, cls=encoder)
             assert json_str == json_str_2
 
+class TestNumpyGenericCodec(TestNumpyCoding):
+    def setup_method(self):
+        self.codec = NPY_DTYPE_CODEC
+        self.objs = [np.float16(1.0), np.float32(1.0), np.float64(1.0),
```

Comment:
How odd, for np.float64;

`{':is_custom:': True, '__class__': 'float64', '__module__': 'numpy', 'dtype': 'float64', 'bytes': b'\x00\x00\x00\x00\x00\x00\xf0?'}`

Just json.dumps to 1.0.

Reply:
Ok so the problem is effectively based on this: https://github.com/python/cpython/issues/84655#issuecomment-1093870644

The default method is only called _if_ the dict can't be serialised, and unfortunately numpy.float64 is JSON serializable to a standard float.

Do we actually care? If not we can just have a bit of a lossy conversion.

Pinging @dwhswenson specifically.